core/block_svg.js

```diff
@@ -124,16 +130,18 @@ Blockly.BlockSvg.INLINE = -1;
  */
 Blockly.BlockSvg.prototype.initSvg = function() {
   goog.asserts.assert(this.workspace.rendered, 'Workspace is headless.');
-  // Input shapes are empty holes drawn when a value input is not connected.
-  for (var i = 0, input; input = this.inputList[i]; i++) {
-    input.init();
-    if (input.type === Blockly.INPUT_VALUE) {
-      this.initInputShape(input);
+  if (!this.isInsertionMarker()) { // Insertion markers not allowed to have inputs or icons
```

Comment:
hmm, I think maybe the comment is misleading. the inputs will still exist, so statement inputs will still be rendered. but they aren't initialized, so fields on them aren't generated/initialized, and their SVG elements are not created, as desired: https://github.com/LLK/scratch-blocks/blob/develop/core/input.js#L208